bin/set_version

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Add a new version to the version file. The version file is a tab separated list of version numbers
+# and git commit hashes in reverse order (newest is at the top of the file). The version may contain
+# a short description as well.
+
+import argparse
+import datetime
+import os
+import pathlib
+import subprocess
+
+pdo_source_root=pathlib.Path(__file__).parent.parent
+version_file = pdo_source_root / 'VERSION'
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    '--version-file', '-f',
+    help='File where version information is stored',
+    type=str)
+
+parser.add_argument(
+    '--version', '-v',
+    help='Version to commit, form X.Y.Z for major, minor and patch versions',
+    type=str)
+
+parser.add_argument(
+    '--description', '-d',
+    help='Optional description to add to the version',
+    type=str)
+
+parser.add_argument(
+    '--commit', '-c',
+    help='Git hash of the commit to associate with the version',
+    type=str)
+
+options = parser.parse_args()
+
+# Get the VERSION file and current source path
+if options.version_file :
+    version_file = pathlib.Path(options.version_file)
+    pdo_source_root = version_file.parent
+
+# Compute the new version, by default the version will increment the minor version of the most
+# recent version and reset the patch level. the version is of the form x.y.z, there may be an
+# optional 'v' at the beginning of the version string
+if options.version :
+    version = options.version
+    try :
+        (major, minor, patch) = version.strip('v').split('.')
+    except :
+        print(f'Invalid version string; {version}')
+        sys.exit(1)
+
+else :
+    # get the current version information from the version file
+    with open(version_file, 'r') as vf :
+        (version, commit, *rest) = vf.readline().strip().split('\t')
+
+    try :
+        (major, minor, patch) = version.strip('v').split('.')
+    except :
+        print(f'Invalid version string; {version}')
+        sys.exit(1)
+
+    minor = int(minor) + 1
+    patch = 0
+
+# Compute the commit to associate with the new version
+if options.commit :
+    command = ['git', 'rev-parse', options.commit]
+    output = subprocess.run(command, cwd=pdo_source_root, capture_output=True, text=True)
```

Comment:
we should check exit code and abort on error. Right now if you provide a non-existing commit (or tag) with `-c` the version file gets updated with `<non-existing-commit>   <today's-date>`

Reply:
will add this

Reply:
Seems issue is that there is a missing `, check=True ` among the subprocess.run arguments to get an exception. Once you add that it does the right thing ...